agent/command/taskdata_send.go

```diff
@@ -1,75 +0,0 @@
-package command
-
-import (
-	"context"
-	"os"
-
-	"github.com/evergreen-ci/evergreen/agent/internal"
-	"github.com/evergreen-ci/evergreen/agent/internal/client"
-	"github.com/evergreen-ci/utility"
-	"github.com/mitchellh/mapstructure"
-	"github.com/pkg/errors"
-)
-
-type taskDataSend struct {
-	File     string `mapstructure:"file" plugin:"expand"`
-	DataName string `mapstructure:"name" plugin:"expand"`
-	base
-}
-
-func taskDataSendFactory() Command   { return &taskDataSend{} }
-func (c *taskDataSend) Name() string { return "json.send" }
-
-func (c *taskDataSend) ParseParams(params map[string]interface{}) error {
-	if err := mapstructure.Decode(params, c); err != nil {
-		return errors.Wrapf(err, "error decoding '%v' params", c.Name())
-	}
-
-	if c.File == "" {
-		return errors.New("'file' param must not be blank")
-	}
-
-	if c.DataName == "" {
-		return errors.New("'name' param must not be blank")
-	}
-
-	return nil
-}
-
-func (c *taskDataSend) Execute(ctx context.Context,
-	comm client.Communicator, logger client.LoggerProducer, conf *internal.TaskConfig) error {
-
-	td := client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret}
-
-	errChan := make(chan error)
-	go func() {
-		// attempt to open the file
-		fileLoc := getJoinedWithWorkDir(conf, c.File)
-		jsonFile, err := os.Open(fileLoc)
-		if err != nil {
-			errChan <- errors.Wrap(err, "Couldn't open json file")
-			return
-		}
-
-		jsonData := map[string]interface{}{}
-		err = utility.ReadJSON(jsonFile, &jsonData)
-		if err != nil {
-			errChan <- errors.Wrap(err, "File contained invalid json")
-			return
-		}
-
-		errChan <- errors.Wrapf(comm.PostJSONData(ctx, td, c.DataName, jsonData),
```

Comment:
Seems like `PostJSONData`, [the legacy REST v1 route](https://github.com/evergreen-ci/evergreen/blob/976819115af360a38710e7ba65ec9dedd21ac7dd/service/api.go#L814), and [the newer REST v2 route](https://github.com/evergreen-ci/evergreen/blob/976819115af360a38710e7ba65ec9dedd21ac7dd/rest/route/service.go#L94) could be removed since they're now unused.